packages/scaffold/src/views/w3m-register-account-name-view/index.ts

```diff
@@ -0,0 +1,188 @@
+    return html`
+      <wui-flex
+        flexDirection="column"
+        alignItems="center"
+        gap="m"
+        .padding=${['0', 's', 'm', 's'] as const}
+      >
+        <form ${ref(this.formRef)} @submit=${this.onSubmitName.bind(this)}>
+          <wui-ens-input
+            .disabled=${this.loading}
+            @inputChange=${this.onNameInputChange.bind(this)}
+            .errorMessage=${this.error}
+            .value=${this.name}
+          >
+          </wui-ens-input>
+          ${this.submitButtonTemplate()}
+          <input type="submit" hidden />
+        </form>
+        ${this.templateSuggestions()}
+      </wui-flex>
+    `
+  }
+
+  // -- Private ------------------------------------------- //
+  private submitButtonTemplate() {
+    const showSubmit = this.isAllowedToSubmit()
+
+    return showSubmit
+      ? html`
+          <wui-icon-link
+            size="sm"
+            icon="chevronRight"
+            iconcolor="accent-100"
+            @click=${this.onSubmitName.bind(this)}
+          >
+          </wui-icon-link>
+        `
+      : null
+  }
+
+  private onDebouncedNameInputChange = CoreHelperUtil.debounce((value: string) => {
+    if (EnsController.validateName(value)) {
+      this.error = ''
+      this.name = value
+      EnsController.getSuggestions(value)
+      EnsController.isNameRegistered(value).then(registered => {
+        this.registered = registered
+      })
+    } else if (value.length < 4) {
+      this.error = 'Name must be at least 4 characters long'
+    } else {
+      this.error = 'Can only contain letters, numbers and - characters'
+    }
+  })
+
+  private onSelectSuggestion(name: string) {
+    return () => {
+      this.name = name
+      this.registered = false
+      this.requestUpdate()
+    }
+  }
+
+  private onNameInputChange(event: CustomEvent<string>) {
+    this.onDebouncedNameInputChange(event.detail)
+  }
+
+  private nameSuggestionTagTemplate() {
+    if (this.loading) {
+      return html`<wui-loading-spinner size="lg" color="fg-100"></wui-loading-spinner>`
+    }
+
+    return this.registered
+      ? html`<wui-tag variant="shade" size="lg">Registered</wui-tag>`
+      : html`<wui-tag variant="success" size="lg">Available</wui-tag>`
+  }
+
+  private templateSuggestions() {
+    if (!this.name || this.name.length < 4 || this.error) {
+      return null
+    }
+
+    const suggestions = this.registered ? this.suggestions : []
+
+    return html`<wui-flex flexDirection="column" gap="xxs" alignItems="center">
+      <wui-flex
+        .padding=${['m', 'm', 'm', 'm'] as const}
+        justifyContent="space-between"
+        class="suggestion"
+      >
+        <wui-text color="fg-100" variant="paragraph-400" class="suggested-name">
+          ${this.name}</wui-text
+        >${this.nameSuggestionTagTemplate()}
+      </wui-flex>
+      ${suggestions
+        .filter(s => s.name !== this.name)
+        .map(suggestion => this.availableNameTemplate(suggestion.name))}
```

Comment:
I think even though it's a simple filtering, makings any filter/sort/reduce on outside of the render would be a nice approach to keep things more readable

```suggestion
const filteredNames = suggestions.filter(s => s.name !== this.name)

      ${filteredNames.map(suggestion => this.availableNameTemplate(suggestion.name))}
```